src/trace/span-inferrer.spec.ts

```diff
@@ -53,6 +57,107 @@ describe("SpanInferrer", () => {
     });
   });
 
+  it("remaps all SNS inferred span service name based on DD_SERVICE_MAPPING", () => {
+    // Set the environment variable that depicts test outcome, i.e. the result of service tag
+    process.env.DD_SERVICE_MAPPING = "lambda_sns:new-name";
+    const inferrer = new SpanInferrer(mockWrapper as unknown as TracerWrapper);
+    inferrer.createInferredSpan(snsEvent, {} as any, {} as SpanContext);
+
+    let modifiedSnsEvent = JSON.parse(JSON.stringify(snsEvent));
+    modifiedSnsEvent.Records[0].EventSubscriptionArn = "arn:aws:sns:us-east-1:123456789012:DifferentTopic";
+    inferrer.createInferredSpan(modifiedSnsEvent, {} as any, {} as SpanContext);
+
+    expect(mockWrapper.startSpan).toHaveBeenNthCalledWith(1, "aws.sns", {
+      childOf: {},
+      startTime: 1643039127968,
+      tags: {
+        _inferred_span: { synchronicity: "async", tag_source: "self" },
+        event_subscription_arn: "arn:aws:sns:us-east-1:123456789012:ExampleTopic",
+        message_id: "95df01b4-ee98-5cb9-9903-4c221d41eb5e",
+        operation_name: "aws.sns",
+        request_id: undefined,
+        "resource.name": "ExampleTopic",
+        resource_names: "ExampleTopic",
+        service: "new-name",
+        "span.type": "sns",
+        subject: "example subject",
+        topic_arn: "arn:aws:sns:us-east-1:123456789012:ExampleTopic",
+        topicname: "ExampleTopic",
+        type: "Notification",
+      },
+    });
+    expect(mockWrapper.startSpan).toHaveBeenNthCalledWith(2, "aws.sns", {
+      childOf: {},
+      startTime: 1643039127968,
+      tags: {
+        _inferred_span: { synchronicity: "async", tag_source: "self" },
+        event_subscription_arn: "arn:aws:sns:us-east-1:123456789012:DifferentTopic",
+        message_id: "95df01b4-ee98-5cb9-9903-4c221d41eb5e",
+        operation_name: "aws.sns",
+        request_id: undefined,
+        "resource.name": "ExampleTopic",
+        resource_names: "ExampleTopic",
+        service: "new-name",
+        "span.type": "sns",
+        subject: "example subject",
+        topic_arn: "arn:aws:sns:us-east-1:123456789012:ExampleTopic",
+        topicname: "ExampleTopic",
+        type: "Notification",
+      },
+    });
+  });
+
+  it("remaps specific SNS inferred span service name based on DD_SERVICE_MAPPING topicname", () => {
+    // Set the environment variable that depicts test outcome, i.e. the result of service tag
+    process.env.DD_SERVICE_MAPPING = "DifferentTopic:new-name";
+    console.log(process.env.DD_SERVICE_MAPPING);
+    const inferrer = new SpanInferrer(mockWrapper as unknown as TracerWrapper);
+    inferrer.createInferredSpan(snsEvent, {} as any, {} as SpanContext);
+
+    let modifiedSnsEvent = JSON.parse(JSON.stringify(snsEvent));
+    modifiedSnsEvent.Records[0].Sns.TopicArn = "arn:aws:sns:us-east-1:123456789012:DifferentTopic";
```

Comment:
Causes a different topicname to be extracted here https://github.com/DataDog/datadog-lambda-js/pull/392/files#diff-c90b7ab89897f78a5515f4fbe787c8b74912175b75a69a58459a1bec936f2498R299

thus we can assert below only the service associated with the topic name `DifferentTopic` is remapped to new-name